src/flint/test/test_all.py

```diff
@@ -2995,65 +3038,89 @@ def quick_poly():
 
         if is_field:
-            assert quick_poly() / 3 == mpoly({(0, 0): S(1, 3), (0, 1): S(2, 3), (1, 0): S(1), (2, 2): S(4, 3)})
+            if (P is flint.fmpz_mod_mpoly or P is flint.nmod_mpoly):
+                assert quick_poly() / 3 == mpoly({(0, 0): S(34), (0, 1): S(68), (1, 0): S(1), (2, 2): S(35)})
+            else:
+                assert quick_poly() / 3 == mpoly({(0, 0): S(1, 3), (0, 1): S(2, 3), (1, 0): S(1), (2, 2): S(4, 3)})
         else:
             assert raises(lambda: quick_poly() / 3, DomainError)
 
-        assert 1 // quick_poly() == P(ctx=ctx)
-        assert 1 % quick_poly() == P(1, ctx=ctx)
-        assert divmod(1, quick_poly()) == (P(ctx=ctx), P(1, ctx=ctx))
-
-        assert raises(lambda: quick_poly() / None, TypeError)
-        assert raises(lambda: quick_poly() // None, TypeError)
-        assert raises(lambda: quick_poly() % None, TypeError)
-        assert raises(lambda: divmod(quick_poly(), None), TypeError)
-
-        assert raises(lambda: None / quick_poly(), TypeError)
-        assert raises(lambda: None // quick_poly(), TypeError)
-        assert raises(lambda: None % quick_poly(), TypeError)
-        assert raises(lambda: divmod(None, quick_poly()), TypeError)
-
-        assert raises(lambda: quick_poly() / 0, ZeroDivisionError)
-        assert raises(lambda: quick_poly() // 0, ZeroDivisionError)
-        assert raises(lambda: quick_poly() % 0, ZeroDivisionError)
-        assert raises(lambda: divmod(quick_poly(), 0), ZeroDivisionError)
-
-        assert raises(lambda: 1 / P(ctx=ctx), ZeroDivisionError)
-        assert raises(lambda: 1 // P(ctx=ctx), ZeroDivisionError)
-        assert raises(lambda: 1 % P(ctx=ctx), ZeroDivisionError)
-        assert raises(lambda: divmod(1, P(ctx=ctx)), ZeroDivisionError)
-
-        assert raises(lambda: quick_poly() / P(ctx=ctx), ZeroDivisionError)
-        assert raises(lambda: quick_poly() // P(ctx=ctx), ZeroDivisionError)
-        assert raises(lambda: quick_poly() % P(ctx=ctx), ZeroDivisionError)
-        assert raises(lambda: divmod(quick_poly(), P(ctx=ctx)), ZeroDivisionError)
-
-        assert raises(lambda: quick_poly() / P(1, ctx=ctx1), IncompatibleContextError)
-        assert raises(lambda: quick_poly() // P(1, ctx=ctx1), IncompatibleContextError)
-        assert raises(lambda: quick_poly() % P(1, ctx=ctx1), IncompatibleContextError)
-        assert raises(lambda: divmod(quick_poly(), P(1, ctx=ctx1)), IncompatibleContextError)
-
         f = mpoly({(1, 1): 4, (0, 0): 1})
         g = mpoly({(0, 1): 2, (1, 0): 2})
-        assert f * g / mpoly({(0, 1): 1, (1, 0): 1}) \
-            == mpoly({(1, 1): 8, (0, 0): 2})
-
-        if not is_field:
-            assert raises(lambda: 1 / quick_poly(), DomainError)
-            assert raises(lambda: quick_poly() / P(2, ctx=ctx), DomainError)
+        if (P is flint.fmpz_mod_mpoly or P is flint.nmod_mpoly) and not ctx.is_prime():
+            pass
+        else:
```

Comment:
They do not when `ctx.is_prime` is false, division is not even attempted there. The type check is just to short-circuit because the others don't have an `is_prime` method.

Reply:
But should there not be some tests in the `pass` block?

Like `assert raises(..., lambda: quick_poly() / 2)`?

Also presumably at least the `TypeError` cases below apply equally whether the modulus is prime or not.